sys/include/ztimer.h

```diff
@@ -474,8 +474,103 @@ ztimer_now_t _ztimer_now_extend(ztimer_clock_t *clock);
 /**
  * @brief   Get the current time from a clock
  *
- * @warning don't compare ztimer_now() values from different clocks. The
- *          clocks are almost certainly not synchronized.
+ * There are several caveats to consider when using values returned by
+ * `ztimer_now()` (or comparing those values to results of @ref ztimer_set,
+ * which are compatible unless MODULE_ZTMIER_NOW64 is in use):
+ *
+ * * A single value has no meaning of its own. Meaningful results are only ever
+ *   produced when subtracting values from each other (in the wrapping fashion
+ *   implied by the use of unsigned integers in C).
+ *
+ *   For example, even though it may be the case in some scenarios, the value
+ *   does **not** indicate time since system startup.
+ *
+ * * Only values obtained from the same clock can be compared.
+ *
+ * * Two values can only be compared when the clock has been continuously
+ *   active between the first and the second reading.
+ *
+ *   A clock is guaranteed to be active from the time any timer is set (the
+ *   first opportunity to get a "now" value from it is the return value of @ref
+ *   ztimer_set) until the time the timer's callback returns. The clock also
+ *   stays active when timers are set back-to-back (which is the case when the
+ *   first timer's callback sets the second timer), or when they overlap (which
+ *   can be known by starting the second timer and afterwards observing that
+ *   @ref ztimer_is_set or @ref ztimer_remove returns true in a low-priority
+ *   context).
+ *
+ *   In contrast, the clock is not guaranteed to be active if a timer is
+ *   removed and then a second one is started (even if the thread does not
+ *   block between these events), or when an expiring timer wakes up a thread
+ *   that then sets the second timer.
+ *
+ *   If the clock was active, then the difference between the second value and
+ *   the first is then the elapsed time in the clock's unit, **modulo the
+ *   clock's wrap-around time**.
```

Comment:
This is always `2**32-1`.

Reply:
if you do not jump the 0 or ff..ff then it should be mod 2^32

<!-- going with 4 Bit 0..15 
start: 2 end: 7 234567  7-2  5count
start: b(11) end: 2  bcdef012 2-11=-9 +15  ZN 0+N=0 Z15 -->

Reply:
I'm not using the precise value due to the presence of the ZTIMER_NOW64 module; if you think that can be disregarded here I'm happy to add that in.

Reply:
On second thought (and re-reading the text), I'm putting 2^32 in there with a note on ZTIMER_NOW64 (so that in particular if the latter gets deprecated / removed, the text automatically becomes better usable).